### ISSUE TITLE
Receiving `Using z.date() within output schema is forbidden.` error, but using `ez.dateOut()` already

### ISSUE THREAD

**Issue by topherauyeung** [RobinTail/express-zod-api#966]
Hi! First - fantastic job with this library, it's been a delight to work with so far.

I recently tried generating some OpenAPI documentation for the first time and ran into the following issue:
```
A [OpenAPIError]: Using z.date() within output schema is forbidden. Please use z.dateOut() instead. Check out the documentation for details.
```
but all of my inputs/outputs are already defined using `ez.dateIn()` and `ez.dateOut()`. Grepping my repo for `z.date()` results in no matches. I'm using pretty much the exact same configuration for the yaml generation as is in the documentation:
```
  new OpenAPI({
    routing,
    config,
    version: '0.0.1',
    title: 'Flock API',
    serverUrl: 'https://flockhealth.com',
    composition: 'inline'
  }).getSpecAsYaml();
```

Any ideas?

**Comment by RobinTail** [comment]
Hello @topherauyeung ,

thank you for reaching out.

There is a schema walker under the hood that iterates your routing and schemas and calling depicting functions depending on its `._def.typeName`. The error message you get is located inside `depictDate` function and is only possible to be thrown in `schema._def.typeName` is `ZodDate`. Perhaps you have some custom schema that inherits `ZodDate` somehow, or you're instantiating `z.date()` with some arguments that make grepping unsuccessful.

Consider the following:

1. Inspect the tracing logs of the thrown error. The library contains sourcemaps and perhaps it could reveal from which function it's getting into `depictDate()`.
2. Consider instantiating `new OpenAPI()` with a customized `routing` parameter where you gonna be adding your endpoints manually one by one, between the attempts of generating the documentation. This will help you to identify the problematic endpoint.

Meanwhile I'll try to improve the informativeness of `OpenAPIError` to eliminate the confusion.

Let me know if you find something.

**Comment by RobinTail** [comment]
I managed to recall that I have a branch where I began working on making those error more informative.

**Comment by RobinTail** [comment]
By the way, @topherauyeung , please also check the schema used in your ResultHandler.

**Comment by RobinTail** [comment]
I'm about to release a new version that might help you to identify the reason for that error,  @topherauyeung

**Comment by RobinTail** [comment]
🚀 More informative error messages released in version 10.5.0.
Please check it out, @topherauyeung , and let me know if it's helpful.

**Comment by topherauyeung** [comment]
Ah! Thanks so much. I actually think I may know the issue now that you've pointed me towards custom types. I am using a Prisma-To-Zod schema generator for some of my output types and I'm betting it's coming from there. Will report back when I investigate tomorrow!
